File: R/getcoef.R
getcoef <- function(fd) {
  	#  Extracts the coefficient array from a functional data object FD, or,
  	#    if FD is a basis object, assigns the identity matrix as the coefficient
  	#    array

  	#  Last modified 10 Feb 2003
  
  	if (inherits(fd, "fd")) {
    	coef <- as.array(fd$coefs)
    	if (length(dim(coef)) == 1) coef <- matrix(coef, length(coef), 1)
  	}  else {
    	if (inherits(fd, "basis.fd")) {
      		coef <- diag(rep(1,fd$nbasis))
    	} else {
      		stop("An object of class fd or basis.fd expected, but not found.")
    	}
  	}
  	return(coef)
}

File: inst/Matlab/fdaM/@fd/getcoef.m
function coefmat = getcoef(fdobj)
%  GETCOEF   Extracts the coefficient array from FDOBJ.
%    If FDOBJ is a basis object, it assigns the identity matrix as the
%    coefficient array.

%  last modified 20 July 2006

  if isa_fd(fdobj) || isa_basis(fdobj)
    fnames = fieldnames(fdobj);
    switch fnames{1}
      case 'coef'
        coefmat = fdobj.coef;
      case 'type'
        basisobj = fdobj.basisobj;
        nbasis = getnbasis(basisobj);
        coefmat = eye(nbasis);
      otherwise
        error('Structure is neither of fd type or of basis type');
    end
  else
    error(['Argument is neither a functional data object', ...
           ' nor a functional basis object.']);
  end

